.buildkite/upload_artifacts.py

```diff
@@ -40,57 +44,95 @@
 # Installer artifact location
 INSTALLER_DIR = os.path.join(PROJECT_PATH, "installer")
 
-
-def create_github_comment(artifacts):
+headers = {'Authorization': 'token %s' % ACCESS_TOKEN}
+
+# Manifest of files, keyed by extension
+file_manifest = {
+    'exe': {
+        'extension': 'exe',
+        'description': 'Windows Installer',
+        'category': 'installer',
+        'content_type': 'application/x-ms-dos-executable',
+    },
+    'pex': {
+        'extension': 'pex',
+        'description': 'Pex file',
+        'category': 'Python package',
+        'content_type': 'application/octet-stream',
+    },
+    'whl': {
+        'extension': 'whl',
+        'description': 'Whl file',
+        'category': 'Python package',
+        'content_type': 'application/zip',
+    },
+    'zip': {
+        'extension': 'zip',
+        'description': 'Zip file',
+        'category': 'Python package',
+        'content_type': 'application/zip',
+    },
+    'gz': {
+        'extension': 'gz',
+        'description': 'Tar file',
+        'category': 'Python package',
+        'content_type': 'application/gzip',
+    },
+    'apk': {
+        'extension': 'apk',
+        'description': 'Android Installer',
+        'category': 'installer',
+        'content_type': 'application/vnd.android.package-archive',
+    },
+}
+
+file_order = [
+    'exe',
+    'apk',
+    'pex',
+    'whl',
+    'zip',
+    'gz',
+]
+
+session = requests.Session()
+
+def create_status_report_html(artifacts):
+    """
+    Create html page to list build artifacts for linking from github status.
+    """
+    html = "<html>\n<body>\n<h1>Build Artifacts</h1>\n"
+    current_heading = None
+    for ext in file_order:
+        artifact = artifacts[ext]
+        if artifact['category'] != current_heading:
+            current_heading = artifact['category']
+            html += "<h2>{heading}</h2>\n".format(heading=current_heading)
+        html += "<p>{description}: <a href='{media_url}'>{name}</a></p>\n".format(
+            **artifact
+        )
+    html += "</body>\n</html>"
+    return html
```

Comment:
Yeah, I didn't use templates because I didn't want (and also was not 100% sure how) to add dependencies to the build process.